File: sources/com/s8/stack/web/carbide/objform/ScalarObjFormSetter.java
package com.s8.stack.web.carbide.objform;

import com.s8.io.bohr.neon.core.NeBranch;


/**
 * 
 * @author pierreconvert
 *
 */
public class ScalarObjFormSetter extends PrimitiveObjFormSetter {

	public ScalarObjFormSetter(NeBranch branch) {
		super(branch, "/s8-stack-web/carbide/objform/ScalarObjFormGetter");
	}
	
	
	public void setUnit(String unit) {
		vertex.setStringUTF8("unit", unit);
	}

}

File: web-sources/s8-stack-web/carbide/objform/ScalarObjFormSetter.js


import { PrimtiveObjFormSetter } from '/s8-stack-web/carbide/objform/PrimtiveObjFormSetter.js';


export class ScalarObjFormSetter extends PrimtiveObjFormSetter {


    /**
     * 
     */
    value = "(unset)";

    constructor() {
        super(2);
    }

    createInputNode() {

        /* <input> */
        let inputWrapperNode = document.createElement("div");
        inputWrapperNode.classList.add("objform-input");
        this.inputNode = document.createElement("input");
        this.inputNode.setAttribute("type", "text");
        inputWrapperNode.appendChild(this.inputNode);
        this.fieldNode.appendChild(inputWrapperNode);

        let _this = this;
        this.inputListener = function(){ _this.sendValue(); };
        this.inputNode.addEventListener("blur", this.inputListener);
        /* </input> */

        /* <unit> */
        let unitWrapperNode = document.createElement("div");
        unitWrapperNode.classList.add("objform-unit");
        this.unitNode = document.createElement("span");
        this.unitNode.innerHTML = "(unit)";
        unitWrapperNode.appendChild(this.unitNode);
        this.fieldNode.appendChild(unitWrapperNode);
        /* </unit> */



    }


    sendValue(){
        let newInputValue = this.inputNode.value;
        if(newInputValue != this.value){
            this.value = newInputValue;
            this.S8_vertex.runFloat32("set-value", this.value);
        }
    }

    setUnit(abbreviation){
        this.unitNode.innerHTML = abbreviation;
    }


    S8_set_unit(abbreviation){
        this.setUnit(abbreviation);
    }

    S8_set_value(value){
        this.inputNode.value = value;
    }

}
